### PR TITLE
Textarea UI component

### LINKED ISSUE #12684: Scrollable text box for use with the template rendering system

### PR DESCRIPTION
**Pull request by georgewrmarshall**
Fixes: #12684 

Explanation:  Textarea UI component

**TODO**
- [x] unit testing
- [x] confirm font-size. Our other inputs are 16px. Design is 14px. Have set to 16px currently

Manual testing steps:  
  - Go to most recent build of storybook
  - Search for `Textarea`
  - Test controls
  
Images
Screen Shot 2021-11-12 at 4 24 46 PM

Screen Shot 2021-11-12 at 4 25 28 PM

Screen Shot 2021-11-12 at 4 26 23 PM

### CODE REVIEW

File: ui/components/ui/textarea/README.mdx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';
+
+import TextArea from '.';
+
+# TextArea
+
+TextAreas communicate actions that users can take.
```
Comment (georgewrmarshall) [inline ui/components/ui/textarea/README.mdx:7]: Think this is left over from the copied Button docs page should be something like:
```suggestion
TextArea allows users to enter text into the UI
```

File: .prettierignore [hunk 2]
```diff
@@ -9,4 +9,4 @@ app/vendor/**
 .nyc_output/**
 .vscode/**
 test/e2e/send-eth-with-private-key-test/**
-*.scss
+# *.scss
```
Comment (georgewrmarshall) [inline .prettierignore:12]: Sorry I missed what the decision was to do on this? I thought we were just going to remove from the PR?

**Review by georgewrmarshall — COMMENTED** [review]
One update to description of docs page. That was my bad.

File: .prettierignore (hunk 2 above)
Comment (hmalik88) [reply .prettierignore:12]: Yes, going to remove -- thanks for the reminder!

File: ui/components/ui/textarea/index.scss [hunk 3]
```diff
@@ -0,0 +1,27 @@
+@use "design-system";
+
+.textarea {
+  display: block;
+  box-shadow: none;
+  color: design-system.$black;
+
+  // not sure why this mixin isn't working: @include H6;
```
Comment (darkwing) [inline ui/components/ui/textarea/index.scss:8]: Should we resolve this issue before merging?

Comment (hmalik88) [reply ui/components/ui/textarea/index.scss:8]: Ah yes, the mixin works in other components so I'm not exactly sure why it's failing to register here. Going to look into this a bit more.

File: ui/components/ui/textarea/index.scss [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  font-size: 1rem;
+  line-height: 140%;
+  font-style: normal;
+  font-weight: normal;
```
Comment (georgewrmarshall) [inline ui/components/ui/textarea/index.scss:12]: Ok so if `@use` is defined I think it overrides the usual imports and you need to traverse `design-system`

```suggestion
  @include design-system.H6;
```

File: ui/components/ui/textarea/index.scss (hunk 3 above)
Comment (georgewrmarshall) [reply ui/components/ui/textarea/index.scss:8]: See comment below

**Review by georgewrmarshall — COMMENTED** [review]
Looking good. Figured out the mixin issue too

File: ui/components/ui/textarea/index.scss [hunk 5]
```diff
@@ -0,0 +1,25 @@
+@use "design-system";
+
+.textarea {
+  display: block;
+  box-shadow: none;
+  color: design-system.$black;
+
+  @include design-system.H6;
+
+  font-size: 1rem;
```
Comment (georgewrmarshall) [inline ui/components/ui/textarea/index.scss:10]: The mixin should take care of `font-size` `design-system.H6 = 14px` `1rem = 16px`
```suggestion
  @include design-system.H6;
```

It's 14px in the design.
Screen Shot 2021-11-24 at 3 16 24 PM

**Review by georgewrmarshall — COMMENTED** [review]
Not sure if my comment was a mistake and missed the font-size

File: ui/components/ui/textarea/index.scss (hunk 5 above)
Comment (hmalik88) [reply ui/components/ui/textarea/index.scss:10]: Yup, I left that there on purpose. I checked with @holantonela yesterday and she said font should be 16px

**Review by georgewrmarshall — COMMENTED** [review]
Updates @hmalik88 made look good to me!

File: ui/components/ui/textarea/textarea.test.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    };
+  });
+  it('should render the TextArea component without crashing', () => {
+    const { getByText } = render(<TextArea {...args} />);
+    expect(getByText(text)).toBeDefined();
+  });
+  it('should call onChange when there is a change made', () => {
+    const { container } = render(<TextArea {...args} />);
+    fireEvent.change(container.firstChild, { target: { value: 'abc' } });
+    expect(onChange).toHaveBeenCalled();
```
Comment (adonesky1) [inline ui/components/ui/textarea/textarea.test.js:39]: [nit] I think we ought to test that the input text is rendered rather than whether the onChange method was called

**Review by darkwing — DISMISSED** [review]

File: ui/css/design-system/attributes.scss [hunk 7]
```diff
@@ -50,36 +33,20 @@ $fractions: (
...
+$sizes-strings: xs, sm, md, lg, xl, none;
 
 $border-style: solid, double, none, dashed, dotted;
 $directions: top, right, bottom, left;
 $display: block, grid, flex, inline-block, inline-grid, inline-flex, list-item;
 $text-align: left, right, center, justify, end;
 $font-weight: bold, normal, 100, 200, 300, 400, 500, 600, 700, 800, 900;
 $font-style: normal, italic, oblique;
-$font-size: 10px, 12px;
+$font-size: 12px;
```
Comment (adonesky1) [inline ui/css/design-system/attributes.scss:49]: why was this change required?

**Review by adonesky1 — DISMISSED** [review]
small non-blocking questions. Otherwise LGTM

File: ui/components/ui/textarea/textarea.test.js (hunk 6 above)
Comment (hmalik88) [reply ui/components/ui/textarea/textarea.test.js:39]: Good suggestion, will make a note to make a PR for this in the future!

File: ui/css/design-system/attributes.scss (hunk 7 above)
Comment (georgewrmarshall) [reply ui/css/design-system/attributes.scss:49]: Hmm apologies, I can't remember maybe just revert. This is being used on the Typography component with the `fontSize` prop but it probably shouldn't exist. That's what the `variant` prop is for... but that's a different discussion

**Review by adonesky1 — APPROVED** [review]

**Review by darkwing — APPROVED** [review]
